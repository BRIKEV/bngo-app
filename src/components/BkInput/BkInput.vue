<template>
  <div class="input-container">
    <input
      :data-cy="dataCy"
      :value="value"
      :name="name"
      :type="type"
      :placeholder="placeholder"
      :id="id"
      required
      :class="color"
      @input="$emit('input', $event.target.value)"
    />
    <label :for="id" :class="color">{{ label }}</label>
  </div>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'BkInput',
  props: {
    dataCy: VueTypes.string,
    value: VueTypes.string,
    name: VueTypes.string,
    type: VueTypes.string,
    label: VueTypes.string,
    placeholder: VueTypes.string,
    required: VueTypes.bool,
    color: VueTypes.string,
    id: VueTypes.string,
  },
};
</script>
<style lang="scss" scoped>
  @import "@/theme/index.scss";
  .input-container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    width: 100%;
    position: relative;
    margin: calculateRem(30px) 0;
    input {
      font-size: $fs-small;
      border: none;
      width: 100%;
      outline: none;
      background: none;
      height: calculateRem(50px);
      &:active ~label,
      &:focus ~label,
      &:valid ~label {
        bottom: calculateRem(50px);
      }
      &.primary {
        color: $brand;
        border-bottom: calculateRem(2px) solid $white;
      }
      &.secundary {
        color: black;
        border-bottom: calculateRem(2px) solid black;
      }
      &:-webkit-autofill,
      &:-webkit-autofill:hover,
      &:-webkit-autofill:focus {
        background-color: transparent;
      }
      &:focus::placeholder {
        color:transparent;
      }
      &::placeholder {
        text-align: right;
      }
    }
    label {
      color: $white;
      bottom: calculateRem(12px);
      position: absolute;
      font-size: $fs-large;
      font-weight: 700;
      transition: all .3s ease;
      &.primary {
        color: $white;
      }
      &.secundary {
        color: black;
      }
    }
  }
</style>
