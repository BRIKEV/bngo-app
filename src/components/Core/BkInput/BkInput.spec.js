import { mount } from '@vue/test-utils';
import BkInput from '@/components/Core/BkInput/BkInput.vue';

describe('BkInput component', () => {
  it('Should be render correctly', () => {
    const wrapper = mount(BkInput);
    expect(wrapper.element).toMatchSnapshot();
  });
});
