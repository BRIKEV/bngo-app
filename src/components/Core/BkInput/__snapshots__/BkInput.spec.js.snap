// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BkInput component Should be render correctly 1`] = `
<div
  class="input-container"
>
  <input
    class=""
    data-cy=""
    id=""
    name=""
    placeholder=""
    required="required"
    type=""
  />
   
  <label
    class=""
    for=""
  >
    
  </label>
</div>
`;
